src/modules/form/components/DynamicField.tsx

```diff
@@ -97,6 +104,16 @@ const DynamicField: React.FC<DynamicFieldProps> = ({
       itemChanged({ linkId, value, type: ChangeType.User }),
     [linkId, itemChanged]
   );
+  const onChangeFileValue = useCallback(
```

Comment:
It might be a little nearer of the Uploader component's `onChange` had different types depending on the `multiple` value. For individual uploader, the onChange would just be a single file (or string), but for multi-upload it would be an array. That's how the select (MUI Autocomplete) works, which can also handle both single or multiple selection. With that approach we wouldn't need this wrapper, I don't think.

Reply:
I couldn't manage to get this working with typescript happy ([here's](https://github.com/greenriver/hmis-frontend/pull/982/commits/7693f65121fe4bcb09b0607752c12ea34fa05967) my attempt - maybe there is something simple I am not seeing), but I managed something similar with two different type definitions. Let me know if it looks okay to you.